src/execution_algorithms/kfac/kfac_util.cpp

```diff
@@ -430,6 +430,66 @@ void allgather_blocks(
   }
 }
 
+template <El::Device Device>
+void allgather_inverse_matrices_sizes(
```

Comment:
Can we compute the inverse matrix sizes independently on each proc during the setup stage? This approach incurs an extra blocking communication at every KFAC step.

Reply:
We are not communicating inverse matrix sizes in the current pipeline.  This function will be used when we cannot calculate matrix sizes and will be called only in the first step.